jumanji/environments/swarms/search_and_rescue/env.py

```diff
@@ -0,0 +1,381 @@
+        searcher_view_angle: float = 0.75,
+        time_limit: int = 400,
+        viewer: Optional[Viewer[State]] = None,
+        target_dynamics: Optional[TargetDynamics] = None,
+        generator: Optional[Generator] = None,
+        reward_fn: Optional[RewardFn] = None,
+        observation: Optional[ObservationFn] = None,
+    ) -> None:
+        """Instantiates a `SearchAndRescue` environment
+
+        Args:
+            target_contact_range: Range at which a searchers will 'find' a target.
+            searcher_max_rotate: Maximum rotation searcher agents can
+                turn within a step. Should be a value from [0,1]
+                representing a fraction of pi radians.
+            searcher_max_accelerate: Maximum acceleration/deceleration
+                a searcher agent can apply within a step.
+            searcher_min_speed: Minimum speed a searcher agent can move at.
+            searcher_max_speed: Maximum speed a searcher agent can move at.
+            searcher_view_angle: Searcher agent local view angle. Should be
+                a value from [0,1] representing a fraction of pi radians.
+                The view cone of an agent goes from +- of the view angle
+                relative to its heading, e.g. 0.5 would mean searchers have a
+                90° view angle in total.
+            time_limit: Maximum number of environment steps allowed for search.
+            viewer: `Viewer` used for rendering. Defaults to `SearchAndRescueViewer`.
+                target_dynamics:
+            target_dynamics: Target object dynamics model, implemented as a
+                `TargetDynamics` interface. Defaults to `RandomWalk`.
+            generator: Initial state `Generator` instance. Defaults to `RandomGenerator`
+                with 20 targets and 10 searchers.
+            reward_fn: Reward aggregation function. Defaults to `SharedRewardFn` where
+                agents share rewards if they locate a target simultaneously.
+        """
+        # self.searcher_vision_range = searcher_vision_range
+        self.target_contact_range = target_contact_range
+
+        self.searcher_params = AgentParams(
+            max_rotate=searcher_max_rotate,
+            max_accelerate=searcher_max_accelerate,
+            min_speed=searcher_min_speed,
+            max_speed=searcher_max_speed,
+            view_angle=searcher_view_angle,
+        )
+        self.time_limit = time_limit
+        self._target_dynamics = target_dynamics or RandomWalk(0.001)
+        self.generator = generator or RandomGenerator(num_targets=100, num_searchers=2)
+        self._viewer = viewer or SearchAndRescueViewer()
+        self._reward_fn = reward_fn or SharedRewardFn()
+        self._observation = observation or AgentAndTargetObservationFn(
+            num_vision=64,
+            vision_range=0.1,
+            view_angle=searcher_view_angle,
+            agent_radius=0.01,
+            env_size=self.generator.env_size,
```

Comment:
In this case do we want to pass these arguments in from the constructor arguments to make tweaking these values a bit a more streamlined the arguments are standardised across the different vision models? Though appreciate this is a standard pattern.

I did this when testing with Mava, passing in the type and constructing here.

Reply:
Ye it can be quite a mission to instantiate, it's possible we should reconsider this pattern, but for now leave it as is just to stay consistent